### ISSUE TITLE
SetTimeout on NodeList

### ISSUE THREAD

**Issue by sanandnarayan** [Codebrahma/blockchain#8]
https://github.com/Codebrahma/blockchain/blob/master/util/nlist.js#L14

The right way to do this is to clearTimeout whenever you receive a heartbeat and setTimeout after 60s

**Comment by vrajeshkanna** [comment]
This would require one timer per node/peer opposed to one per wallet which we have now. Why do we node one timer per node?

**Comment by aalavandhan** [comment]
|-----(10s)-----------|---------------------(65s)---------------------| => node 1
---------|-----------------------------------|-------------------------  => setInterval 

Consider this example, node1 was inactive for more than 60s and thus should have been unregistered. However a global timer check, will miss this. 

It makes more sense to check for activity after 'n' seconds on each node after a heartbeat.

**Comment by vrajeshkanna** [comment]
Yes. I thought through this problem. worst case possibility as we are checking for > is 120 seconds. Is it not already solved by this.delay/2. Cleanup of nodes which have heartbeats older than 60 sec is done every 30 secs. Even we can make it 10 seconds(this.delay/6) as it is done locally, compared to having n timers  where n is the number of nodes it would be a linear complexity. Having n timers would increase the complexity by n.
